caom2pipe/data_source_composable.py

```diff
@@ -234,3 +235,58 @@ def get_time_box_work(self, prev_exec_time, exec_time):
                 "ORDER BY ingestDate ASC "
         self._logger.debug(query)
         return mc.query_tap_client(query, self._client)
+
+
+@dataclass
+class StateRunnerMeta:
+    entry_name: str  # how to refer to the item of work to be processed
+    entry_ts: datetime.timestamp  # timestamp associated with item of work
+
+
+class QueryTimeBoxDataSourceTS(DataSource):
+    """
+    Implements the identification of the work to be done, by querying a
+    TAP service, in time-boxed chunks. The time values are timestamps
+    (floats).
+
+    Deprecate the QueryTimeBoxDataSouce class in favour of this implementation.
+    """
+
+    def __init__(self, config, preview_suffix='jpg'):
+        super(QueryTimeBoxDataSourceTS, self).__init__(config)
+        self._preview_suffix = preview_suffix
+        subject = mc.define_subject(config)
+        self._client = CadcTapClient(subject, resource_id=self._config.tap_id)
+        self._logger = logging.getLogger(__name__)
```

Comment:
Is there a best practice for declaring loggers at CADC? I would prefer (and I changed the code to do, but I can change it back)
self._logger = logging.getLogger(self.__class__.__name__), because that is the most informative in a log file.

But if there's a better/more accepted way to handle this, let me know, and I will change it.

Reply:
From what I've seen global on the file is more common but I don't have a strong opinion on this.